### PR TITLE
chore: add github workflow for tests

### PR DESCRIPTION
**Pull request by Sebastian-Haehnlein**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      - name: Install Packages
+        run: |
+          npm ci
+          cd e2e
+          npm ci
+
+      - name: Code Quality
+        run: |
+          npm run format
+          bash .gitlab-ci-test-no-changes.sh 'you probably committed unformatted code'
```
Comment (dhhyi) [inline .github/workflows/test.yml]: rename script to something general without 'gitlab'

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+          cd e2e
+          npx tsc -p cypress/tsconfig.json
+
+      - name: Lint
+        run: |
+          npm run lint
+
+      - name: Check TSLint no Warnings
+        run: |
+          sh .gitlab-ci-check-no-additional-warnings.sh
```
Comment (dhhyi) [inline .github/workflows/test.yml]: rename script to something general without 'gitlab'

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+          npm i -g wait-on
+          npm ci
+          cd e2e
+          npm ci
+
+      - name: Test Jest
+        run: |
+          npm test
+
+      - name: Test E2E mocked
```
Comment (dhhyi) [inline .github/workflows/test.yml]: use https://github.com/cypress-io/github-action here

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        run: |
+          npm run format
+          bash .gitlab-ci-test-no-changes.sh 'you probably committed unformatted code'
+          node reports/tslint-gitlab
+
+      - name: Compile
+        run: |
+          npx tsc -p tsconfig.spec.json
+          cd e2e
+          npx tsc -p cypress/tsconfig.json
```
Comment (dhhyi) [inline .github/workflows/test.yml]: `ng build --aot` should also be performed

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          path: ~/.cache
+          key: ${{ runner.os }}-quality-cache-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-quality-cache-
+
+      - name: Install Packages
+        run: |
+          npm ci
+          cd e2e
+          npm ci
```
Comment (dhhyi) [inline .github/workflows/test.yml]: e2e `npm install` is not needed here, right?

File: .github/workflows/test.yml (hunk 2 above)
Comment (dhhyi) [reply .github/workflows/test.yml]: also: integrate into previous step?

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+      - name: Compile
+        run: |
+          npx tsc -p tsconfig.spec.json
+          cd e2e
+          npx tsc -p cypress/tsconfig.json
+
+      - name: Lint
+        run: |
+          npm run lint
```
Comment (dhhyi) [inline .github/workflows/test.yml]: how about https://github.com/mooyoul/tslint-actions?
Then comments seems to be added on the PR

File: .github/workflows/test.yml (hunk 3 above)
Comment (Sebastian-Haehnlein) [reply .github/workflows/test.yml]: tested, but the stability of the test and the advantage itself is not really given

File: .github/workflows/test.yml (hunk 6 above)
Comment (Sebastian-Haehnlein) [reply .github/workflows/test.yml]: results in an updated tslint.json (relative paths)

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        run: |
+          npm run ng -- build --aot
+          npx tsc -p tsconfig.spec.json
+          cd e2e
+          npx tsc -p cypress/tsconfig.json
+
+      - name: Lint
+        run: |
+          npm run lint
+          sh scripts/ci-check-no-additional-warnings.sh
```
[the remark this answers is not in the document]
Comment (Sebastian-Haehnlein) [reply .github/workflows/test.yml]: this script shows disabled tests 

e.g.
WARNING: /home/my/projects/intershop/github/intershop/intershop-pwa/e2e/cypress/integration/specs/account/edit-user-email.b2c.e2e-spec.ts:47:3 - Disabled test.

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        run: |
+          npm ci
+          cd e2e
+          npm ci
+
+      - name: Code Quality
+        run: |
+          npm run format
+          bash scripts/ci-test-no-changes.sh 'you probably committed unformatted code'
+          node reports/tslint
```
[the remark this answers is not in the document]
Comment (Sebastian-Haehnlein) [reply .github/workflows/test.yml]: "node reports/tslint" or "sh scripts/ci-check-no-additional-warnings.sh" should be present to show warning for disabled tests.

File: .github/workflows/test.yml [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+          npm run lint
+          sh scripts/ci-check-no-additional-warnings.sh
+
+      - name: Lint by action
+        uses: mooyoul/tslint-actions@v1.1.1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          pattern: '*.ts'
+          config: 'tslint.json'
+          project: './'
```
Comment (dhhyi) [inline .github/workflows/test.yml]: please try with `token` and `project` only, the `config` value defaults to `tslint.json`, the value for `project` should be `tsconfig.json`, i think then the Disabled Tests are also listed

File: .github/workflows/test.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (dhhyi) [reply .github/workflows/test.yml]: see: https://github.com/intershop/intershop-pwa/pull/10/files#r350688700

File: .github/workflows/test.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (dhhyi) [reply .github/workflows/test.yml]: see: https://github.com/intershop/intershop-pwa/pull/10/files#r350688700

**Comment by dhhyi** [comment]
replaced by #20

**Pull request closed without merging**